drivers/include/periph/qspi.h

```diff
@@ -0,0 +1,234 @@
+} qspi_erase_size_t;
+#endif
+
+/**
+ * @brief   Configuration Options for QSPI
+ * @{
+ */
+#define QSPI_FLAG_1BIT          (0x0)   /**< single data line           */
+#define QSPI_FLAG_2BIT          (0x1)   /**< two parallel data lines    */
+#define QSPI_FLAG_4BIT          (0x2)   /**< four parallel data lines   */
+#define QSPI_FLAG_8BIT          (0x3)   /**< eight parallel data lines  */
+
+#define QSPI_FLAG_BIT_MASK      (0x3)   /**< mask to get data lines from flags */
+#define QSPI_FLAG_BIT(flags)    ((flags) & QSPI_FLAG_BIT_MASK)
+
+#define QSPI_FLAG_DDR           (0x4)   /**< use Double Data Rate mode  */
+
+#define QSPI_FLAG_ADDR_32BIT    (0x8)   /**< use 32 bit addressing      */
+/** @} */
+
+/**
+ * @brief   Basic initialization of the given QSPI bus
+ *
+ * @note    This function MUST not be called more than once per bus!
+ *
+ * @param[in] bus       QSPI device to initialize
+ */
+void qspi_init(qspi_t bus);
+
+/**
+ * @brief   Configure the QSPI peripheral settings
+ *
+ * @param[in] bus       QSPI device to configure
+ * @param[in] mode      QSPI mode @see qspi_mode_t
+ * @param[in] flags     QSPI Configuration Options
+ * @param[in] clk_hz    QSPI frequency in Hz
+ */
+void qspi_configure(qspi_t bus, qspi_mode_t mode, uint32_t flags, uint32_t clk_hz);
+
+/**
+ * @brief   Start a new QSPI transaction
+ *
+ * Starting a new QSPI transaction will get exclusive access to the QSPI bus
+ * and configure it according to the given values. If another QSPI transaction
+ * is active when this function is called, this function will block until the
+ * other transaction is complete (qspi_relase was called).
+ *
+ * @param[in] bus       QSPI device to access
+ */
+void qspi_acquire(qspi_t bus);
+
+/**
+ * @brief   Finish an ongoing QSPI transaction by releasing the given QSPI bus
+ *
+ * After release, the given SPI bus should be fully powered down until acquired
+ * again.
+ *
+ * @param[in] bus       QSPI device to release
+ */
+void qspi_release(qspi_t bus);
+
+/**
+ * @brief   Execute a command with response data e.g Read Status, Read JEDEC
+ *
+ * @param[in]  bus      QSPI device
+ * @param[in]  command  The JEDEC command ID
+ * @param[out] response Command response data, may be NULL
+ * @param[in]  len      Size of the command response buffer
+ */
+void qspi_cmd_read(qspi_t bus, uint8_t command, void *response, size_t len);
+
+/**
+ * @brief   Execute a command with data e.g Write Status, Write Enable
+ *
+ * @param[in] bus       QSPI device
+ * @param[in] command   The JEDEC command ID
+ * @param[in] data      Command parameter data, may be NULL
+ * @param[in] len       Size of the command response buffer
+ */
+void qspi_cmd_write(qspi_t bus, uint8_t command, const void *data, size_t len);
+
+/**
+ * @brief   Read data from external memory.
+ *          Typically it is implemented by quad read command (`0x6B`).
+ *
+ * @param[in]  bus      QSPI device
+ * @param[in]  address  Address to read from
```

Comment:
The nrf52 expects the address to be word-aligned. Is that the same with the same54? Should that be enforced at API-level?

Reply:
That seems to be a limitation of the nrf52.
I'd say the driver can best deal with that :wink: